GitUI/Plugin/PluginRegistry.cs

```diff
@@ -13,26 +15,44 @@ public static class PluginRegistry
 
         public static List<IRepositoryHostPlugin> GitHosters { get; } = new List<IRepositoryHostPlugin>();
 
-        public static void Initialize()
+        public static bool PluginsRegistered { get; private set; }
+
+        /// <summary>
+        /// Initialises all available plugins on the background thread.
+        /// </summary>
+        /// <param name="postInitialiseAsync">A function to execute once plugins are loaded.</param>
+        public static Task InitializeAsync(Func<Task> postInitialiseAsync)
         {
             lock (Plugins)
             {
                 if (Plugins.Count > 0)
                 {
-                    return;
+                    return Task.CompletedTask;
                 }
 
-                foreach (var plugin in ManagedExtensibility.GetExports<IGitPlugin>().Select(lazy => lazy.Value))
+                try
                 {
-                    plugin.SettingsContainer = new GitPluginSettingsContainer(plugin.Name);
-
-                    if (plugin is IRepositoryHostPlugin repositoryHostPlugin)
+                    foreach (var plugin in ManagedExtensibility.GetExports<IGitPlugin>().Select(lazy => lazy.Value))
                     {
-                        GitHosters.Add(repositoryHostPlugin);
-                    }
+                        plugin.SettingsContainer = new GitPluginSettingsContainer(plugin.Name);
+
+                        if (plugin is IRepositoryHostPlugin repositoryHostPlugin)
+                        {
+                            GitHosters.Add(repositoryHostPlugin);
+                        }
 
-                    Plugins.Add(plugin);
+                        Plugins.Add(plugin);
+                    }
+                }
+                catch
+                {
+                    // no-op
                 }
+
+#pragma warning disable VSTHRD105 // Avoid method overloads that assume TaskScheduler.Current
```

Comment:
❗️ This should be fixed rather than suppressed. You can make the containing method asynchronous, and then simply put the flag initialization in a `finally` block.